kclvm/tools/src/query/override.rs

```diff
@@ -138,44 +339,235 @@ impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
+        entry: &mut ast::ConfigEntry,
     ) -> (Vec<String>, Vec<String>) {
-        (vec![], vec![])
+        let mut paths = vec![];
+        let mut paths_with_id = vec![];
+        if let Some(key) = &entry.key {
+            let path = match &key.node {
+                ast::Expr::Identifier(identifier) => identifier.get_name(),
+                ast::Expr::StringLit(string_lit) => string_lit.value.clone(),
+                _ => return (paths, paths_with_id),
+            };
+            paths.push(path.clone());
+            paths_with_id.push(path.clone());
+            let (value_paths, value_paths_with_id) = self.get_field_paths(&mut entry.value);
+            if !value_paths.is_empty() {
+                paths.append(
+                    &mut value_paths
+                        .iter()
+                        .map(|value_path| format!("{}.{}", path, value_path))
+                        .collect::<Vec<String>>(),
+                );
+                paths_with_id.append(
+                    &mut value_paths_with_id
+                        .iter()
+                        .map(|value_path| format!("{}|{}", path, value_path))
+                        .collect::<Vec<String>>(),
+                );
+            }
+        }
+        (paths, paths_with_id)
     }
-    pub(crate) fn _find_schema_config_and_repalce(
+
+    /// Get config key path from the AST key node.
+    #[inline]
+    pub(crate) fn get_path_from_key(&mut self, key: &Option<ast::NodeRef<ast::Expr>>) -> String {
+        match key {
+            Some(key) => match &key.node {
+                ast::Expr::Identifier(identifier) => identifier.get_name(),
+                ast::Expr::StringLit(string_lit) => string_lit.value.clone(),
+                _ => "".to_string(),
+            },
+            None => "".to_string(),
+        }
+    }
+}
+
+#[derive(Debug)]
+enum OverrideConfig<'a> {
+    Schema(&'a mut ast::SchemaExpr),
+    Config(&'a mut ast::ConfigExpr),
+}
+
+impl OverrideTransformer {
+    pub(crate) fn find_schema_config_and_repalce(
         &mut self,
-        _schema_config: &mut ast::SchemaExpr,
-        _field_path: &str,
-        _value: &ast::NodeRef<ast::Expr>,
+        schema_expr: &mut ast::SchemaExpr,
     ) -> bool {
-        false
+        let (paths, paths_with_id) = self.get_schema_config_field_paths(schema_expr);
+        match paths.iter().position(|r| r == &self.field_path) {
+            Some(pos) => {
+                let mut config = OverrideConfig::Schema(schema_expr);
+                self.replace_with_id_path(&mut config, &paths_with_id[pos]);
+                true
+            }
+            None => false,
+        }
+    }
+
+    pub(crate) fn replace_with_id_path(&mut self, config: &mut OverrideConfig, path_with_id: &str) {
+        if path_with_id.is_empty() {
+            return;
+        }
+        let parts = path_with_id.split('|').collect::<Vec<&str>>();
+        for (i, part) in parts.iter().enumerate() {
```

Comment:
The nested for and if-else judgment is too deep here, we'd better to reorganize the code

Reply:
> The nested for and if-else judgment is too deep here, we'd better to reorganize the code

Simplifies the implementation logic of this function by splitting the function